### PR TITLE
feat: config networking

### PR DESCRIPTION
**Pull request by TranQuangPhi**

### CODE REVIEW

**Review by chungnguyenhoang — APPROVED** [review]

File: template/src/constants/interface/redux/UserInterface.ts [hunk 1]
```diff
@@ -14,7 +15,7 @@ export interface IUser {
 export interface IUserActions {
   userLogin: IActionDispatch
   userSignUp: IActionDispatch
-  logout: IActionDispatch
+  logout: ActionCreatorWithoutPayload<any>
```
Comment (hoangSTS) [inline template/src/constants/interface/redux/UserInterface.ts]: Do not use 'any'

File: template/src/services/networking/index.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  url,
+  method,
+  data,
+  config,
+}: {
+  url: string
+  method: string
+  data?: T | string
+  config?: AxiosRequestHeaders
+}): Promise<AxiosResponse<T>> {
```
Comment (anhle10051996) [inline template/src/services/networking/index.ts]: I think we should separate Type to another place:
Ex: 
type Param <`T`> = {
 url: string, 
 ...
}
=> async function axiosAPI <`T`> (params: Param <`T`> ): Promise<AxiosResponse <`T`> >

**Review by anhle10051996 — CHANGES_REQUESTED** [review]

File: template/src/constants/interface/redux/UserInterface.ts (hunk 1 above)
Comment (TranQuangPhi) [reply template/src/constants/interface/redux/UserInterface.ts]: done

File: template/src/services/networking/index.ts (hunk 2 above)
Comment (TranQuangPhi) [reply template/src/services/networking/index.ts]: done

**Review by hoangSTS — APPROVED** [review]

File: template/App.tsx [hunk 3]
```diff
@@ -3,6 +3,7 @@ import {LogBox, Text} from 'react-native';
 import {SafeAreaProvider} from 'react-native-safe-area-context';
 import {Provider} from 'react-redux';
 import MainLayout from './src/MainLayout';
+import { injectStore } from './src/services/networking/axios';
```
Comment (ngochuyduong) [inline template/App.tsx]: no spaces between curly braces

File: template/src/constants/index.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  success: [200, 201],
+  unauthorized: [401, 403],
+}
+
+export const TOKEN_TYPE: ITokenType = {
+  Bearer: 'Bearer',
+  Basic: 'Basic',
+}
+
+export const AXIOS_TIMEOUT: number = 6000
```
Comment (ngochuyduong) [inline template/src/constants/index.ts:18]: Can remove :number

File: template/src/services/networking/axios.ts [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+      logout()
+    })
+}
+
+instance.interceptors.request.use(
+  (config: AxiosRequestConfig) => {
+    // Do something before request is sent
+    return config
+  },
+  error => {
```
Comment (ngochuyduong) [inline template/src/services/networking/axios.ts]: Add type for error

**Review by ngochuyduong — CHANGES_REQUESTED** [review]

File: template/src/services/networking/axios.ts [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+import {userActions} from '../../store/reducers'
+import {RootState} from '../../store/store'
+import {getData, setData, clearAllData} from '../../utilities/storage'
+import {AUTH_API} from '../api/api'
+
+let store: Store<RootState, AnyAction>
+
+const instance = axios.create({
+  baseURL: Config.API_URL,
+  timeout: AXIOS_TIMEOUT,
```
Comment (loc-nguyenthien) [inline template/src/services/networking/axios.ts:14]: This is not like what written in the docs above

File: template/src/services/networking/index.ts [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+  try {
+    if (body?.constructor !== FormData) {
+      throw new Error('Unrecognized FormData part')
+    }
+    const headers: AxiosRequestHeaders = {
+      'content-type': 'multipart/form-data',
+    }
+    const customConfig = {
+      ...config,
+      ...headers,
```
Comment (loc-nguyenthien) [inline template/src/services/networking/index.ts]: If this `headers` only have one value above, why don't we just add it here?

**Review by loc-nguyenthien — CHANGES_REQUESTED** [review]

File: template/src/constants/index.ts (hunk 4 above)
Comment (TranQuangPhi) [reply template/src/constants/index.ts:18]: I defined this type to be in sync with all constant variables

File: template/App.tsx (hunk 3 above)
Comment (TranQuangPhi) [reply template/App.tsx]: done

File: template/src/services/networking/axios.ts (hunk 5 above)
Comment (TranQuangPhi) [reply template/src/services/networking/axios.ts]: done

File: template/src/services/networking/axios.ts (hunk 6 above)
Comment (TranQuangPhi) [reply template/src/services/networking/axios.ts:14]: done

File: template/src/services/networking/index.ts (hunk 7 above)
Comment (TranQuangPhi) [reply template/src/services/networking/index.ts]: done

**Review by huydosgtech — APPROVED** [review]

**Review by anhle10051996 — APPROVED** [review]

**Review by anhle10051996 — APPROVED** [review]

**Review by loc-nguyenthien — APPROVED** [review]

**Review by ngochuyduong — APPROVED** [review]

**Pull request merged**